### PR TITLE
#34 Set up Node.js REST API empty project

### PR DESCRIPTION
**Pull request by bryan936M**
#### What does this PR do?
- create the basic components of a restful api
#### Description of Task to be completed?
- add project structures components(model, routes, test, config, .github, etc) 
- add documentation(swagger)
- configure eslintr
#### How should this be manually tested?
- clone repo
- open terminal & run ' npm install' command
- again; execute 'npm run dev' command
- open browser & search 'http://localhost:3000/' & 'http://localhost:3000/api-docs'
#### Any background context you want to provide?
- set indent space of vscode editor to 2 spaces in settings under 'FILE: EOL' for eslint
- default port is 3000
#### What are the relevant pivotal tracker/Trello stories?
card Id: #34
card link: https://trello.com/c/PHkEazmg

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  env: {
+    browser: true,
+    commonjs: true,
+    es2021: true,
+    node: true,
+  },
+  extends: 'airbnb-base',
```
Comment (chanceDira) [inline .eslintrc.js]: Here let's also add "prettier"

File: app.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+// Imports
+const express = require('express');
```
Comment (chanceDira) [inline app.js]: can you use import instead of require

File: server.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// Import app
+const app = require('./app');
+
+const PORT = process.env.PORT || 3000;
+
+// Listen to port ( default: 3000 )
+app.listen(PORT, () => {
+  console.log('[Server] On');
```
Comment (chanceDira) [inline server.js]: Include the listening PORT

**Review by chanceDira — CHANGES_REQUESTED** [review]
Good job @Bryan, just added some feedback for you

**Review by faid-terence — COMMENTED** [review]
This is very good Bryan , i would also advice you to add a controller folder for better practices
otherwise all is Good, Thanks

**Review by mukunzidd — CHANGES_REQUESTED** [review]
- Please remove the brackets in the title of the PR
- Link your troll card in the description. Not just a card id
- Add a lint script in package.json

Other than that, it looks good.

**Review by faid-terence — APPROVED** [review]

**Review by chanceDira — APPROVED** [review]
Good job @bryan

**Review by chanceDira — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  "homepage": "https://github.com/atlp-rwanda/ecommerce-app-predators-bn#readme",
+  "dependencies": {
+    "body-parser": "^1.20.2",
+    "cors": "^2.8.5",
+    "express": "^4.18.2",
+    "morgan": "^1.10.0",
+    "swagger-jsdoc": "^6.2.8",
+    "swagger-ui-express": "^4.6.2"
+  },
+  "devDependencies": {
```
Comment (ALVINdimpos) [inline package.json:36]: add this dependencies     "@babel/cli": "^7.21.0",
    "@babel/core": "^7.21.3",
    "@babel/preset-env": "^7.20.2",
    "jest": "^29.5.0"

Comment (bryan936M) [reply package.json:36]: can you give me the names of the packages?

**Review by vanessa-abewe — APPROVED** [review]

File: package.json (hunk 4 above)
Comment (nili-alu) [reply package.json:36]: npm install --save-dev @babel/cli
npm install --save-dev @babel/core
npm install @babel/preset-env --save-dev
npm install jest --global for jest,

**Review by nili-alu — APPROVED** [review]

**Review by Muhet — APPROVED** [review]

File: package.json (hunk 4 above)
Comment (ALVINdimpos) [reply package.json:36]: thank you @nili-alu

File: package.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    "url": "git+https://github.com/atlp-rwanda/ecommerce-app-predators-bn.git"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/atlp-rwanda/ecommerce-app-predators-bn/issues"
+  },
+  "homepage": "https://github.com/atlp-rwanda/ecommerce-app-predators-bn#readme",
+  "dependencies": {
```
Comment (ALVINdimpos) [inline package.json:27]: now it's good

**Review by ALVINdimpos — APPROVED** [review]

**Review by faid-terence — APPROVED** [review]

**Review by mukunzidd — APPROVED** [review]

**Pull request merged**